jib-core/src/main/java/com/google/cloud/tools/jib/frontend/ParameterValidator.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.frontend;
+
+import com.google.cloud.tools.jib.builder.BuildLogger;
+import java.util.List;
+import java.util.function.Function;
+
+/** Used for validating configuration parameters. */
+public class ParameterValidator {
+
+  /**
+   * Checks a list for null or empty elements. Warns the user if an empty string is found, and
+   * throws an error if a null element is found.
+   *
+   * @param list the list to check
+   * @param parameterName the name of the configuration parameter being checked
+   * @param logger used for displaying messages
+   * @param exception a function returning the exception to be thrown when a null element is found
+   * @param <T> the exception type for null elements
+   * @throws T if a null element is found
+   */
+  public static <T extends Throwable> void checkListForNullOrEmpty(
```

Comment:
I'm a little confused by this, if we know we are checking for null or empty, then we should be able to return that status back to the caller.

Reply:
and not through an exception?

Reply:
Okay, so I missed the point of this whole thing, it's more of a Preconditions for userConfiguration

Reply:
It looks like I'm talking to myself but... perhaps jib-core should not be in the business of providing utility functions to its users. If we need to enforce something, we should still do it in the core-code.

If the tools built on top of jib-core needs to use something like checkListForNullOrEmpty, they should implement it themselves or pull it from a guava-like library?

I don't see it being a huge issue to have this check duplicated in the maven and gradle plugins.

Reply:
I'm kind of confused about what to do with this PR now. It was originally just supposed to be a simple error message fix, but it's getting a lot of conflicting feedback and the scope has grown quite a bit. I feel like we need to further discuss how much input processing/enforcement jib-core should be responsible for (and possibly close this PR until then; it's getting messy).

Reply:
Could you just change `BuildConfiguration.Builder`'s `setJvmFlags()` and `setArgs()` (or `JibPluginConfiguration`'s `setJvmFlags()` and `setArgs()`) to something like:
```java
    public Builder setJvmFlags(@Nullable List<String> jvmFlags) {
      if (jvmFlags != null) {
        Preconditions.checkArgument(Iterables.all(jvmFlags, Objects::nonNull));
        this.jvmFlags = ImmutableList.copyOf(jvmFlags);
      }
      return this;
    }
```

Reply:
That seems like a good idea to me, it's elegant and enforces the constraints we want without the user-facing guava dependency. I'm still going to close this PR though, I think at this point it's easier to just make a new one than to revert what I need to in this one to implement the change.